Sources/Commands/SwiftPackageTool.swift

```diff
@@ -966,3 +1051,207 @@ fileprivate func logPackageChanges(changes: [(PackageReference, Workspace.Packag
+    struct Directories {
+        let sources: RelativePath
+        let tests: RelativePath?
+        let createSubDirectoryForModule: Bool
+    }
+    
+    enum PackageType: String {
+        case executable
+        case library
+        case systemModule = "system-module"
+        case empty
+        case manifest
+    }
+    
+    let directories: Directories
+    let type: PackageType
+}
+
+extension InitPackage.PackageTemplate {
+    internal init(template: PackageTemplate) throws {
+        
+        let packageType: InitPackage.PackageType
+        switch template.type {
+        case .executable:
+            packageType = .executable
+        case .library:
+            packageType = .library
+        case .systemModule:
+            packageType = .systemModule
+        case .empty:
+            packageType = .empty
+        case .manifest:
+            packageType = .manifest
+        }
+        
+        self.init(sourcesDirectory: template.directories.sources,
+                  testsDirectory: template.directories.tests,
+                  createSubDirectoryForModule: template.directories.createSubDirectoryForModule,
+                  packageType: packageType)
+    }
+}
+
+internal var useNewBehaviour: Mode {
+    get {
+        switch (ProcessEnv.vars["SWIFTPM_ENABLE_PACKAGE_CREATE"].map { $0.lowercased() }) {
+        case "true":
+            return .new
+        default:
+            return .legacy
+        }
+    }
+}
+
+internal enum Mode {
+    case new
+    case legacy
+}
+
+internal func getSwiftPMDefaultTemplate(type: InitPackage.PackageType,
+                                        sources: RelativePath = .init("./Sources"),
+                                        tests: RelativePath? = nil,
+                                        createSubDirectoryForModule: Bool = false) -> PackageTemplate {
+    // Even if we are making a "classic" package that doesn't use a template we should till use templates
+    // for consistency within the codebase
+    let defaultDir = PackageTemplate.Directories(sources: sources, tests: tests, createSubDirectoryForModule: createSubDirectoryForModule)
+    let packageType: PackageTemplate.PackageType
+    
+    switch type {
+    case .executable:
+        packageType = .executable
+    case .library:
+        packageType = .library
+    case .systemModule:
+        packageType = .systemModule
+    case .empty:
+        packageType = .empty
+    case .manifest:
+        packageType = .manifest
+    default:
+        packageType = .library
+    }
+    
+    switch useNewBehaviour {
+    case .new:
+        return PackageTemplate(directories: defaultDir, type: packageType)
+    case .legacy:
+        let legacyDir = PackageTemplate.Directories(sources: RelativePath("./Sources"), tests: RelativePath("./Tests"), createSubDirectoryForModule: true)
+        return PackageTemplate(directories: legacyDir, type: packageType)
+    }
+}
+
+fileprivate func makePackage(fileSystem: FileSystem,
```

Comment:
this function should be be in an extension to InitPackage rather than "free floating" to better encapsulate the logic